fabric-data-generation-api-v1/build.gradle

```diff
@@ -56,62 +56,126 @@ import org.objectweb.asm.ClassReader
 	doLast {
 		File inputJar = loom.namedMinecraftProvider.parentMinecraftProvider.commonJar.toFile()
 		String accessWidener = file("template.accesswidener").text + "\n"
 
-		visitMethods(inputJar, "net/minecraft/data/server/recipe/RecipeProvider.class") { name, desc, owner ->
+		def classes = getClasses(inputJar)
+
+		visitMethods(classes["net/minecraft/data/server/recipe/RecipeProvider"]) { name, desc, owner ->
 			if (it.name == "generate")
 				return
 
 			accessWidener += "transitive-accessible\tmethod\t${owner}\t${name}\t${desc}\n"
 		}
 
-		visitMethods(inputJar, "net/minecraft/data/client/BlockStateModelGenerator.class") { name, desc, owner ->
+		visitMethods(classes["net/minecraft/data/client/BlockStateModelGenerator"]) { name, desc, owner ->
 			if (desc == "()V")
 				// Skip over methods that dont take any arguments, as they are specific to minecraft.
 				return
 
 			accessWidener += "transitive-accessible\tmethod\t${owner}\t${name}\t${desc}\n"
 		}
 
-		visitMethods(inputJar, "net/minecraft/data/server/loottable/BlockLootTableGenerator.class") { name, desc, owner ->
+		visitMethods(classes["net/minecraft/data/server/loottable/BlockLootTableGenerator"]) { name, desc, owner ->
 			accessWidener += "transitive-accessible\tmethod\t${owner}\t${name}\t${desc}\n"
 		}
 
-		visitMethods(inputJar, "net/minecraft/data/client/ItemModelGenerator.class") { name, desc, owner ->
+		visitMethods(classes["net/minecraft/data/client/ItemModelGenerator"]) { name, desc, owner ->
 			accessWidener += "transitive-accessible\tmethod\t${owner}\t${name}\t${desc}\n"
 		}
 
+		classes.values().forEach { classNode ->
+			visitFinalMethods(classNode) { name, desc, owner ->
+				if (name != "getName" || desc != "()Ljava/lang/String;") {
+					// Not the method we are after
+					return
+				}
+
+				if (!hasAncestor(classNode, classes, "net/minecraft/data/DataProvider")) {
+					// Not a decedent of DataProvider
+					return
+				}
+
+				accessWidener += "transitive-extendable\tmethod\t${owner}\t${name}\t${desc}\n"
+			}
+		}
+
 		file("src/main/resources/fabric-data-generation-api-v1.accesswidener").text = accessWidener
 	}
 }
 
-def visitMethods(File input, String className, closure) {
-	def clazz = getClassNode(input, className)
-
-	clazz.methods.forEach {
+def visitMethods(ClassNode classNode, closure) {
+	classNode.methods.forEach {
 		if ((it.access & Opcodes.ACC_SYNTHETIC) != 0 || (it.access & Opcodes.ACC_PUBLIC) != 0)
 			return
 
 		if (it.name.startsWith("<"))
 			return
 
-		closure(it.name, it.desc, clazz.name)
+		closure(it.name, it.desc, classNode.name)
+	}
+}
+
+def visitFinalMethods(ClassNode classNode, closure) {
+	classNode.methods.forEach {
+		if (!Modifier.isFinal(it.access))
+			return
+
+		if (it.name.startsWith("<"))
+			return
+
+		closure(it.name, it.desc, classNode.name)
 	}
 }
 
-ClassNode getClassNode(File input, String className) {
+// Return a map of all class names to classNodes
+def getClasses(File input) {
```

Comment:
Hmm, I wonder how much memory this uses

Reply:
Not enough for me to worry, its only ever manually ran. I should be able to vastly improve this by not reading the code 👍